hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NSSummaryEndpoint.java

```diff
@@ -0,0 +1,1061 @@
+      namespaceSummaryResponse.setNumVolume(volumes.size());
+      List<OmBucketInfo> allBuckets = listBucketsUnderVolume(null);
+      namespaceSummaryResponse.setNumBucket(allBuckets.size());
+      int totalNumDir = 0;
+      long totalNumKey = 0L;
+      for (OmBucketInfo bucket : allBuckets) {
+        long bucketObjectId = bucket.getObjectID();
+        totalNumDir += getTotalDirCount(bucketObjectId);
+        totalNumKey += getTotalKeyCount(bucketObjectId);
+      }
+      namespaceSummaryResponse.setNumTotalDir(totalNumDir);
+      namespaceSummaryResponse.setNumTotalKey(totalNumKey);
+      break;
+    case VOLUME:
+      namespaceSummaryResponse =
+          new NamespaceSummaryResponse(EntityType.VOLUME);
+      List<OmBucketInfo> buckets = listBucketsUnderVolume(names[0]);
+      namespaceSummaryResponse.setNumBucket(buckets.size());
+      int totalDir = 0;
+      long totalKey = 0L;
+
+      // iterate all buckets to collect the total object count.
+      for (OmBucketInfo bucket : buckets) {
+        long bucketObjectId = bucket.getObjectID();
+        totalDir += getTotalDirCount(bucketObjectId);
+        totalKey += getTotalKeyCount(bucketObjectId);
+      }
+      namespaceSummaryResponse.setNumTotalDir(totalDir);
+      namespaceSummaryResponse.setNumTotalKey(totalKey);
+      break;
+    case BUCKET:
+      namespaceSummaryResponse =
+          new NamespaceSummaryResponse(EntityType.BUCKET);
+      assert (names.length == 2);
+      long bucketObjectId = getBucketObjectId(names);
+      namespaceSummaryResponse.setNumTotalDir(getTotalDirCount(bucketObjectId));
+      namespaceSummaryResponse.setNumTotalKey(getTotalKeyCount(bucketObjectId));
+      break;
+    case DIRECTORY:
+      // path should exist so we don't need any extra verification/null check
+      long dirObjectId = getDirObjectId(names);
+      namespaceSummaryResponse =
+          new NamespaceSummaryResponse(EntityType.DIRECTORY);
+      namespaceSummaryResponse.setNumTotalDir(getTotalDirCount(dirObjectId));
+      namespaceSummaryResponse.setNumTotalKey(getTotalKeyCount(dirObjectId));
+      break;
+    case KEY:
+      namespaceSummaryResponse = new NamespaceSummaryResponse(EntityType.KEY);
+      break;
+    case UNKNOWN:
+      namespaceSummaryResponse =
+          new NamespaceSummaryResponse(EntityType.UNKNOWN);
+      namespaceSummaryResponse.setStatus(ResponseStatus.PATH_NOT_FOUND);
+      break;
+    default:
+      break;
+    }
+    return Response.ok(namespaceSummaryResponse).build();
+  }
+
+  /**
+   * DU endpoint to return datasize for subdirectory (bucket for volume).
+   * @param path request path
+   * @param listFile show subpath/disk usage for each key
+   * @param withReplica count actual DU with replication
+   * @return DU response
+   * @throws IOException
+   */
+  @GET
+  @Path("/du")
+  @SuppressWarnings("methodlength")
+  public Response getDiskUsage(@QueryParam("path") String path,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 142 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.